### PR TITLE
Areca SATA RAID Controller support + optimizations

### PR DESCRIPTION
**Pull request by vampik**
Hello.
I've added Areca SATA RAID Controller support.
Please note that SAS RAID Controllers are not supported: different syntax is needed (slot/enclosure number instead of just slot number) and they support much more disks (up to 512) making simple brute-force search not possible (1024 slot/enclosure combinations).
I also had to make some optimizations, which effectively reduced scan time from 18 to 5 seconds in my case. I removed excessive smartctl calls, filtered out non-disk devices (enclosures, dvd drives etc.).

### CODE REVIEW

**Comment by v-zhuravlev** [comment]
Hi ,what is 16?
```
if ( $disk->{subdisk} == 0 and $vendor eq "Areca" and $product eq "RAID controller" ) {
        for (my $i = 1; $i <= 16; $i++) {
            push @disks,
                get_smart_disks(
                    {
                        disk_name => $disk->{disk_name},
                        disk_args => "-d areca,$i",
                        disk_model => '',
                        disk_sn => '',
                        subdisk   => 1
                    }
                );
        }
        return @disks;
    }
```
Is this maximum number of disks supported by SATA RAID controller ?

File: discovery-scripts/nix/smartctl-disks-discovery.pl [hunk 1]
```diff
@@ -103,6 +103,9 @@ sub get_smart_disks {
     $disk->{disk_cmd} = $disk->{disk_name};
     if (length($disk->{disk_args}) > 0){
         $disk->{disk_cmd}.=q{ }.$disk->{disk_args};
+        if ( $disk->{subdisk} == 1 and $disk->{disk_args} =~ /-d\s+[^,\s]+,(\S+)/) {
```
Comment (v-zhuravlev) [inline discovery-scripts/nix/smartctl-disks-discovery.pl]: Can you explain these lines please ?

**Comment by v-zhuravlev** [comment]
Thank you, merged

**Comment by vampik** [comment]
> Is this maximum number of disks supported by SATA RAID controller ?

Yes, exactly.

**Comment by vampik** [comment]
> Can you explain this part please?

Append identifier to the disk name. This turns multiple indistinguishable "/dev/sg0" entries in Zabbix into "/dev/sg0 (0)", "/dev/sg0 (1)" etc.

**Pull request merged**